src/lib/db/segment-store.ts

```diff
@@ -203,7 +203,7 @@ export default class SegmentStore implements ISegmentStore {
             id: row.id,
             name: row.name,
             description: row.description,
-            project: row.segment_project_id,
+            project: row.segment_project_id || '',
```

Comment:
Needed some of this back-and-forth here for consistency and due to the way we're binding the project column to projects: Otherwise if we try to set project as `""` we're not able to, as that project does not exist.
This way we have a simple mapping from `""` to `null` and back.

Reply:
I'm not sure I'm 100% convinced on this one, I think the project should be made optional in all the UI code around segments, but maybe I'm not fully understanding what you mean in this comment